src/pushsource/_impl/model/vms.py

```diff
@@ -74,6 +74,16 @@ class VMIPushItem(PushItem):
     boot_mode = attr.ib(type=BootMode, default=None, validator=optional(in_(BootMode)))
     """Boot mode supported by the image (if known): uefi, legacy, or hybrid (uefi + legacy)."""
 
+    marketplace_title_template = attr.ib(type=str, default=None, validator=optional_str)
+    """The template is of the form used by ``str.format``, with available keywords being all of
+    the documented fields on ``VMIRelease`` and ``AMIRelease`` classes."""
+
+    @property
+    def marketplace_title(self) -> str:
+        if not self.marketplace_title_template:
+            return ""
+        return self.marketplace_title_template.format(**attr.attr.asdict(self.release))
```

Comment:
Minor - the double reference to `attr.attr` is a little weird and inconsistent with what currently exists.

The idea has been that anything used within `attr` is imported to `compat_attr` and then exported again.  That means if we drop support for old python/attr versions, at some point we can just replace `from .. import compat_attr as attr` with `import attr` everywhere and not have to touch anything else.  But it won't work here as the pattern was not followed.

Reply:
I've changed it, PTAL again